### PR TITLE
Improved damage indicator

### PR DESCRIPTION
**Pull request by lebroomer**
Adds a new improved damage indicator with customization possibilities.

Can be merged, when the current themes (designed very badly by me with GIMP) are replaced by goodlooking ones.

### CODE REVIEW

File: gamemodes/terrortown/gamemode/server/sv_damage_indicator.lua [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+util.AddNetworkString("ttt2_damage_received")
+
+hook.Add("EntityTakeDamage","TTT2DamageIndicator",function(ent,dmg)
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/server/sv_damage_indicator.lua]: Maybe add this to the GM function, but at the very end. Maybe even `GM:PlayerTakeDamage`. This makes sure that addons that use the hook and modify the damage are handled correctly.

File: gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+-- @section Damage Indicator
+
+DMGINDICATOR = {}
+DMGINDICATOR.themes = {}
+
+local dmgindicator_enable = CreateClientConVar("ttt_dmgindicator_enable", "1", true)
+local dmgindicator_mode = CreateClientConVar("ttt_dmgindicator_mode", "Default", true)
+local dmgindicator_duration = CreateClientConVar("ttt_dmgindicator_duration", "1.5", true)
+local dmgindicator_maxdamage = CreateClientConVar("ttt_dmgindicator_maxdamage", "50.0", true)
+local dmgindicator_maxalpha = CreateClientConVar("ttt_dmgindicator_maxalpha", "255", true)
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua]: you could do it here like I've done it in the armor system or like it is done in the karma system and add these cvars to the table.

File: gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+DMGINDICATOR.cv.maxalpha = CreateClientConVar("ttt_dmgindicator_maxalpha", "255", true)
+
+local lastDamage = CurTime()
+local damageAmount = 0.0
+local maxDamageAmount = 0.0
+
+local function collectDmgIndicatorTextures()
+	local pathBase = "materials/vgui/ttt/dmgindicator/themes/"
+
+	local materials, _ = file.Find(pathBase .. "*.png", "GAME")
```
Comment (Alf21) [inline gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua]: `, _` isn't needed afaik

File: gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+		DMGINDICATOR.themes[materialName] = Material(materialStringBase .. material)
+	end
+end
+collectDmgIndicatorTextures()
+
+net.Receive("ttt2_damage_received", function()
+	local damageReceived = net.ReadFloat()
+
+	if damageReceived <= 0 then
```
Comment (Alf21) [inline gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua]: put it into one line to keep it simple and support the current coding style

File: gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		return
+	end
+
+	lastDamage = CurTime()
+	maxDamageAmount = math.min(1.0, math.max(damageAmount, 0.0) + damageReceived / DMGINDICATOR.cv.maxdamage:GetFloat())
+	damageAmount = maxDamageAmount
+end)
+
+function GM:HUDPaintBackground()
+	if not DMGINDICATOR.cv.enable:GetBool() then
```
Comment (Alf21) [inline gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua]: one line :D

File: gamemodes/terrortown/gamemode/client/cl_help.lua [hunk 6]
```diff
@@ -507,6 +514,50 @@ function HELPSCRN:CreateCrosshairSettings(parent)
...
+function HELPSCRN:CreateDamageIndicatorSettings(parent)
+	local form = vgui.Create("DForm", parent)
+	form:SetName(GetTranslation("f1_dmgindicator_title"))
+
+	form:CheckBox(GetTranslation("f1_dmgindicator_enable"), "ttt_dmgindicator_enable")
+
+	local dmode = vgui.Create("DComboBox", form)
+	dmode:SetConVar("ttt_dmgindicator_mode")
+
+	for name, _ in pairs(DMGINDICATOR.themes) do
```
Comment (Alf21) [inline gamemodes/terrortown/gamemode/client/cl_help.lua]: `, _` isn't needed.

**Comment by TimGoll** [comment]
I tried something. Do you like it?

image

**Pull request merged**